packages/cadl-ranch-coverage-sdk/src/client.ts

```diff
@@ -61,49 +61,83 @@ export class CadlRanchCoverageOperations {
       },
       blobHTTPHeaders: {
         blobContentType: "application/json; charset=utf-8",
       },
     });
 
-    await this.updateIndex(generatorMetadata.name, generatorMetadata.version);
+    await this.updateIndex(generatorMetadata.name, generatorMetadata.version, generatorMetadata.type);
   }
 
-  public async getLatestCoverageFor(generatorName: string): Promise<ResolvedCoverageReport> {
-    const index = await readJsonBlob<{ version: string }>(
+  public async getLatestCoverageFor(generatorName: string): Promise<ResolvedCoverageReport[]> {
+    const index = await readJsonBlob<{ version: string; types: string[] }>(
       this.#container.getBlockBlobClient(`${generatorName}/index.json`),
     );
 
-    const blobClient = this.#container.getBlockBlobClient(`${generatorName}/reports/${index.version}.json`);
-    const blob = await blobClient.download();
-    const body = await blob.blobBody;
-    const content = await body?.text();
-    const report = content ? JSON.parse(content) : undefined;
-    return {
-      generatorMetadata: {
-        version: blob.metadata?.generatorversion,
-        name: blob.metadata?.generatorname,
-      },
-      ...report,
-    };
+    if (!index.types) {
+      // Compatible with current format, delete later
+      const blobClient = this.#container.getBlockBlobClient(`${generatorName}/reports/${index.version}.json`);
+      const blob = await blobClient.download();
+      const body = await blob.blobBody;
+      const content = await body?.text();
+      const report = content ? JSON.parse(content) : undefined;
+      return [
+        {
+          generatorMetadata: {
+            version: blob.metadata?.generatorversion,
+            name: blob.metadata?.generatorname,
+            type: "azure",
+          },
+          ...report,
+        },
+      ];
+    } else {
+      const results = [];
+      for (const type of index.types) {
+        const blobClient = this.#container.getBlockBlobClient(`${generatorName}/reports/${index.version}/${type}.json`);
+        const blob = await blobClient.download();
+        const body = await blob.blobBody;
+        const content = await body?.text();
+        const report = content ? JSON.parse(content) : undefined;
+        results.push({
+          generatorMetadata: {
+            version: blob.metadata?.generatorversion,
+            name: blob.metadata?.generatorname,
+            type: blob.metadata?.reportType,
+          },
+          ...report,
+        });
+      }
+      return results;
+    }
   }
 
-  private async updateIndex(generatorName: string, version: string) {
-    const blob = this.#container.getBlockBlobClient(`${generatorName}/index.json`);
+  private async updateIndex(generatorName: string, version: string, type: string) {
     const data = {
       version,
+      types: [type],
     };
+
+    const blobClient = this.#container.getBlockBlobClient(`${generatorName}/index.json`);
+    if (await blobClient.exists()) {
```

Comment:
Here if you want to stick with this approach you hsould be using the eTag and if match to make sure nothing else changed that file in the meantime or we might get a race condition that will override it. Which feels likely if you run the 2 set of test in parallel

Reply:
an alternative would be to not contain the type but just use the blob list feature.

Reply:
I don't like etag solution because I don't want upload to fail.

The alternative solution still has race condition problem:
1. Version1.azure upload
2. Version2.azure upload
3. Version2.index update
4. Version1.index update

So I'm thinking sending the coverage files in a single command by making arg `coverageFile` an array, i.e.,  `cadl-ranch upload-coverage --coverageFile xxx1.json xxx2.json --generatorName python --storageAccountName azuresdkcadlranch --generatorVersion 3.2.1` and the mode of the coverage would be stored in the xxx1.json itself.

Reply:
Well you keep retrying that’s the point of etag

Reply:
I have some latest change seems don't have the race problem.